### PR TITLE
feat: publish to websocket subs afer height sync

### LINKED ISSUE #300: delay websocket notification to wait for full indexation of tx

### PR DESCRIPTION
**Pull request by jyeshe**
## What

Notifies subscriptions after the height has been indexed, i.e when the data is available through AeMdw API.

## Why

Fixes #300

## Validation steps

Feature:
`elixir --sname aeternity@localhost -S mix test.integration test/integration/ae_mdw_web/websocket_test.exs:409`

Regression:
`elixir --sname aeternity@localhost -S mix test.integration test/integration/ae_mdw_web/websocket_test.exs`

### CODE REVIEW

File: lib/ae_mdw_web/websocket/broadcaster.ex [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+          {{^key, _}, _} -> true
+        end
+
+      case :ets.select(@subs_target_channels, spec, 1) do
+        :"$end_of_table" -> :ok
+        {[_], _cont} -> broadcast(key, data(ser_tx, "Object", source))
+      end
+    end)
+
+    {:noreply, @no_state}
```
[the remark this answers is not in the document]
Comment (jyeshe) [reply lib/ae_mdw_web/websocket/broadcaster.ex:73]: This way makes explicit and more visible that the Broadcaster state is always empty.

File: lib/ae_mdw_web/websocket/broadcaster.ex [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+  end
+
+  defp broadcast(channel, msg) do
+    Riverside.LocalDelivery.deliver(
+      {:channel, channel},
+      {:text, Poison.encode!(msg)}
+    )
+  end
+
+  defp data(data, sub, source),
```
[the remark this answers is not in the document]
Comment (jyeshe) [reply lib/ae_mdw_web/websocket/broadcaster.ex:152]: refactored the old function, thanks!

File: lib/ae_mdw_web/websocket/socket_handler.ex [hunk 3]
```diff
@@ -37,8 +37,8 @@ defmodule AeWebsocket.Websocket.SocketHandler do
       case AeMdw.Validate.id(target) do
         {:ok, id} ->
           {:ok, pid} = Riverside.LocalDelivery.join_channel(id)
-          Listener.register(pid)
-          Listener.register(self())
+          ChainListener.register(pid)
+          ChainListener.register(self())
```
[the remark this answers is not in the document]
Comment (jyeshe) [reply lib/ae_mdw_web/websocket/socket_handler.ex:41]: The Listener monitors the `SocketHandler` and the Riverside channels in this case. It's part of the existing subscription mechanics that we can refactor later to allow selecting subscription only for the node source, mdw source or both.

**Review by thepiwo — APPROVED** [review]
looks good to me

**Review by sborrazas — APPROVED** [review]

**Pull request merged**